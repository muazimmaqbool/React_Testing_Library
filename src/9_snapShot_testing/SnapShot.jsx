import React, { useState } from "react";

/*
->SnapShot Testing: here code of the file is copied in the another file
    -> the copied code is not always in readable mode

  ->snapshot testing copies code of a file and place it in the another folder automatically

  ->Note: When your app/product is in development mode then avoid snapshot testing:

  ->When to use snapshot testing:
    When you application/software is about to go live/launched and you made some minor changes in some files
    (like changing variable names, etc.) and if you made some wrong changes then application won't work in
    production so before making application live do snapshot testing which will tell you that you have made these changes in the end,
    and if you want to keep these changes then update snapshots and if you dont want these changes then remove them.
    Because of this the bugs in your app will be very minimum/ or no bugs at all


*/

const SnapShot = () => {
  const [data, setData] = useState("");
  return (
    <div>
      <h1>Snapshot Testing</h1>
      <button onClick={() => setData("New Data")}>Update Text</button>
      <h1>{data}</h1>
    </div>
  );
};

export default SnapShot;
