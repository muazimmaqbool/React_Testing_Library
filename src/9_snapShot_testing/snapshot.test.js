import { render } from "@testing-library/react";
import SnapShot from "./SnapShot";

test("Snapshot example",()=>{
    const {container}=render(<SnapShot/>);
    expect(container).toMatchSnapshot();
})

/*when this code is run: it will create a snapshot folder and in logs it will say 1 snapshot written:
 and inside that file it will contain a file with name : filename.test.js.snap, 
*/

//watch at(6:00)
//now after make this snapshot file, if you make any in filde (SnapShot.jsx)(change any text their) and run the app
//it will show you 1 snapshot test case failed, it will show initial their was this text persent now its this text present,
//and if you made this change by yourself then press "u" and it will update the snapshot file and if you made this change by mistake then undo the changes 
//in the file

//so by pressing "u" it will update the snapshot file with resent changes and will not fail the test case again.
//or instead of pressing "u" just delete the snapshot file and folder and when you run again it will run perfectly - but this is wrong method

