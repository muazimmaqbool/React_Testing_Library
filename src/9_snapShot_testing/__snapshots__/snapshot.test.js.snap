// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot example 1`] = `
<div>
  <div>
    <h1>
      Snapshot Testing
    </h1>
    <button>
      Update Text
    </button>
    <h1 />
  </div>
</div>
`;
